### DISCUSSION TITLE
Fehlende IPv6 Adresse in Proxmox/OVA RaspberryMatic Installation

### DISCUSSION THREAD

**Discussion by mb-1408-de** [OpenCCU/OpenCCU#2871] [🇩🇪 Fragen&Antworten] [answered]
Hallo zusammen,

ich betreibe gerade auf einem Raspi3 (ELV-Charly) die RaspberryMatic und erhalte dort auch eine IPv6 Adresse von der Fritzbox. Nun würde ich gerne in eine Proxmox/OVA Installation umziehen. Die Installation und der Config Restore haben auch einwandfrei funktioniert. Merke jetzt aber, dass ich mich zu meinem MQTT-Server, der nur eine V6 Adresse hat nicht mehr verbinden kann. Auf der Console ist ersichtlich, dass keine IPv6 Adresse bezogen wird.

Gibt es diesbezüglich einen Unterschied zwischen dem Raspi Image und der x86 OVA Variante? Wie bekomme ich IPv6 aktiviert?

Die Interface-Config in der VM sieht so aus:
```
root@homematic:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel qlen 1000
    link/ether bc:24:11:cc:dd:ee brd ff:ff:ff:ff:ff:ff
    inet 192.168.178.xx/24 brd 192.168.178.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::be24:11ff:fecc:ddee/64 scope link 
       valid_lft forever preferred_lft forever
```
Viele Grüße,
Markus

**Comment by jens-maus** [comment]
> Gibt es diesbezüglich einen Unterschied zwischen dem Raspi Image und der x86 OVA Variante? Wie bekomme ich IPv6 aktiviert?

Nein, da gibt es keinerlei Unterschied und hier funktioniert das tadellos:

```
:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel qlen 1000
    link/ether 00:0c:29:73:77:xx brd ff:ff:ff:ff:ff:ff
    inet 192.168.5.xx/24 brd 192.168.5.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 2a02:xxxx:xxxx:xxxx:xxx:xxxx:xxxx:xxxx/64 scope global dynamic flags 100 
       valid_lft 6446sec preferred_lft 2846sec
    inet6 fe80::20c:29ff:fe73:77xx/64 scope link 
       valid_lft forever preferred_lft forever
```

D.h. du musst den Fehler vmtl. in deiner Proxmox Installation suchen und ich würde erstmal schauen ob eine standard ubuntu vm z.B. eine valide IPv6 Adresse bekommt oder nicht.

**Comment by mb-1408-de** [comment]
Vielen Dank für die prompte Rückmeldung. Ich bin der Ursache näher gekommen. Es werden nicht alle ip6tables Rules geladen, da das "limit" Modul nicht zur Verfügung steht.

```
ubuntu:~/Desktop$ diff -u homematic.txt homematic-raspi.txt 
--- homematic.txt	2024-10-08 16:37:43.786809889 +0200
+++ homematic-raspi.txt	2024-10-08 16:37:28.114841911 +0200
@@ -24,5 +24,11 @@
 -A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 2 -j ACCEPT
 -A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 3 -j ACCEPT
 -A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 4 -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 128 -m state --state NEW -m limit --limit 15/sec -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 129 -m state --state NEW -m limit --limit 15/sec -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 134 -m hl --hl-eq 255 -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 136 -m hl --hl-eq 255 -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 135 -m hl --hl-eq 255 -j ACCEPT
+-A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 137 -m hl --hl-eq 255 -j ACCEPT
 -A INPUT -j REJECT --reject-with icmp6-port-unreachable
 COMMIT
```

Wenn ich ip6tables deaktiviere und noch mal versuche eine Adresse abzurufen funktioniert es:

```
root@homematic:~# ip6tables -P INPUT ACCEPT
root@homematic:~# ip6tables -F INPUT 
root@homematic:~# killall -q udhcpc
root@homematic:~# . /etc/config/netconfig
root@homematic:~# /sbin/udhcpc -b -t 20 -T 3 -S -x "hostname:${HOSTNAME}" -i eth0
udhcpc: started, v1.36.1
udhcpc: broadcasting discover
udhcpc: broadcasting select for 192.168.178.231, server 192.168.178.1
udhcpc: lease of 192.168.178.231 obtained from 192.168.178.1, lease time 864000
deleting routers
adding dns 192.168.178.1
root@homematic:~# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel qlen 1000
    link/ether xxxxxxxxxxxx brd ff:ff:ff:ff:ff:ff
    inet 192.168.178.231/24 brd 192.168.178.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 xxxxxxxxxxxx/64 scope global dynamic flags 100 
       valid_lft 7098sec preferred_lft 1697sec
    inet6 fe80::xxxxxxxxxxxx/64 scope link 
       valid_lft forever preferred_lft forever
```
Alternativ kann man im GUI die Firewall-Policy auf "Ports offen" stellen. 
Beim Versuch die fehlenden Zeilen im Regelwerk nachzuziehen kommt dieser Fehler:

```
root@homematic:~# ip6tables -A INPUT -p ipv6-icmp -m icmp6 --icmpv6-type 128 -m state --state NEW -m limit --limit 15/sec -j ACCEPT
Warning: Extension limit revision 0 not supported, missing kernel module?
ip6tables: No chain/target/match by that name.
```

Installiert ist die Version: 3.77.7.20240826

Eine Debian VM die an der selben Bridge hängt bekommt per "inet6 auto" eine IPv6 Adresse.

**Reply by jens-maus** [reply, marked as answer]
Danke für diese Information und die sehr gute technische Darlegung. In der Tat war mir das nicht bewusst das hier die LIMIT und HF netfilter match kernel module fehlen und es zu diesen Fehlermeldungen kommt. Das hatte die Firewall bisher so nicht gemeldet bzw. ist mir nicht aufgefallen. Danke also dafür. Ich habe auch bereits hierfür einen bug ticket issue erstellt (https://github.com/jens-maus/RaspberryMatic/issues/2874) und auch schon für den nächsten nightly snapshot diese kernel module hinzugefügt, sodass mit dem nächsten nightly snapshot das problem hoffentlich beseitigt sein sollte.

**Reply by mb-1408-de** [reply]
Freut mich, dass ich helfen konnte. Vielen Dank auch an dich für den erstklassigen Support in diesem Projekt und den umgehenden Bugfix.
